File: solutions/csharp/allergies/1/Allergies.cs
using System;
using System.Collections.Generic;


[Flags]
public enum Allergen : short
{
    Eggs=1,
    Peanuts=2,
    Shellfish=4,
    Strawberries=8,
    Tomatoes=16,
    Chocolate=32,
    Pollen=64,
    Cats=128
};

public class Allergies
{
    Allergen mask;
    public Allergies(int mask)
    {
        this.mask=(Allergen) mask;
    }

    public bool IsAllergicTo(Allergen allergen)
    {
        return (allergen & mask) != 0? true : false;
    }

    public Allergen[] List()
    {
        List<Allergen> al = new List<Allergen>();

        for (int m=1; m<=128; m=m<<1) {
            if (((Allergen) m & mask)!=0) al.Add((Allergen) m);
        }
        return al.ToArray();
    }
}

File: solutions/java/allergies/1/src/main/java/Allergies.java
import java.util.ArrayList;
import java.util.List;

public class Allergies {
    List<Allergen> allergies;
    Allergies(int allergiesIn) {
        allergies = new ArrayList<Allergen>();
        allergiesIn = allergiesIn % 256;
        if (allergiesIn%2 == 1) allergies.add(Allergen.EGGS);
        //System.out.println("allergiesIn="+allergiesIn);
        //System.out.println("allergies="+allergies);
        allergiesIn = allergiesIn / 2;
        if (allergiesIn%2 == 1) allergies.add(Allergen.PEANUTS);
        //System.out.println("allergies="+allergies);
        allergiesIn = allergiesIn / 2;
        if (allergiesIn%2 == 1) allergies.add(Allergen.SHELLFISH);
        allergiesIn = allergiesIn / 2;
        if (allergiesIn%2 == 1) allergies.add(Allergen.STRAWBERRIES);
        allergiesIn = allergiesIn / 2;
        if (allergiesIn%2 == 1) allergies.add(Allergen.TOMATOES);
        allergiesIn = allergiesIn / 2;
        if (allergiesIn%2 == 1) allergies.add(Allergen.CHOCOLATE);
        allergiesIn = allergiesIn / 2;
        if (allergiesIn%2 == 1) allergies.add(Allergen.POLLEN);
        allergiesIn = allergiesIn / 2;
        if (allergiesIn%2 == 1) allergies.add(Allergen.CATS);
    }
    Boolean isAllergicTo(Allergen allergenIn) {
        boolean allergicTo = false;
        for (Allergen allergen : allergies )
            if (allergen == allergenIn) allergicTo = true;
        return(allergicTo);
    }
    List<Allergen> getList () {
        return(allergies);
    }

    int size () {
        return (allergies.size());
    }
}